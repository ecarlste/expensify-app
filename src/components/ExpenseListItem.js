import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses.action';

const ExpenseListItem = ({ id, description, amount, createdAt, removeExpense }) => (
  <div>
    <h3>{description}</h3>
    <p>
      {amount} - {createdAt}
    </p>
    <button onClick={() => removeExpense({ id })}>Remove</button>
  </div>
);

const mapDispatchToProps = {
  removeExpense
};

export default connect(
  null,
  mapDispatchToProps
)(ExpenseListItem);
