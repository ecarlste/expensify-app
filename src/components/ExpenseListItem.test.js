import React from 'react';
import { shallow } from 'enzyme';
import { expenseRent } from '../fixtures/expenses.fixture';
import { ExpenseListItem } from './ExpenseListItem';

test('should render ExpenseListItem', () => {
  const wrapper = shallow(<ExpenseListItem {...expenseRent} />);
  expect(wrapper).toMatchSnapshot();
});
