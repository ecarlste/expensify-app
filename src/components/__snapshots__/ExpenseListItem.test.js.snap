// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<Link
  className="list-item"
  to="/edit/3"
>
  <div>
    <h3
      className="list-item__title"
    >
      Rent
    </h3>
    <span
      className="list-item__subtitle"
    >
      December 27th 1969
    </span>
  </div>
  <h3
    className="list-item__content"
  >
    $1,095.00
  </h3>
</Link>
`;
